source/common/router/router.h

```diff
@@ -574,6 +574,7 @@ class Filter : Logger::Loggable<Logger::Id::router>,
   Upstream::ClusterInfoConstSharedPtr cluster_;
   std::unique_ptr<Stats::StatNameDynamicStorage> alt_stat_prefix_;
   const VirtualCluster* request_vcluster_;
+  const PathStatsConfig* path_stats_config_;
```

Comment:
I'm not sure it's worth it to keep a member variable for this. I think you can just look it up when you need it in the route.

Reply:
It is needed similar to `request_vcluster_` because some of the code that increments metrics does not have access to route (not passed in)